next/docs/wasp/docs/guide/core_concepts/state_manager.md

```diff
@@ -38,78 +38,86 @@ contains the same commitment as state index `n`.
 
-To know, which states are available as snapshots, Wasp periodically (see `snapshots.updatePeriod` parameter) updates the known
-snapshot list: it scans local folder and all the network addresses. In local folder it reads all the files with names that satisfy
-search pattern `*-*.snap`. In each network location Wasp reads all the files listed in `INDEX` file of that location. Wasp reads
-a state index and a commitment from the contents of these files. File names are not used to obtain this information and full
-snapshot files are not (down)loaded on this update. The obtained state index and commitment together with exact address
-of the snapshot are stored in known snapshot list. When snapshot is needed by state manager, the list of known snapshots
-is checked and the attempt to (down)load the snapshot from one of the locations in the list is made. If it fails, another
-location is tried. If obtaining snapshot from all the locations fail, it is assumed that the snapshot is not available.
+If chain is started with empty database (usually, if the database hasn't yet been created or it was deleted), the node checks, if
+it can load a snapshot: it scans local folder and all the network addresses for available snapshot files. In local folder it reads
+all the files with names that satisfy search pattern `*-*.snap`. In each network location Wasp reads all the files listed in `INDEX`
+file of that location. Wasp reads a state index and a commitment from the contents of these files. File names are not used to obtain
+this information and full snapshot files are not (down)loaded yet. Among all the available snapshot files, the node chooses the one
+with largest state index and loads it to the store. If there are several files with the same largest state index, the node tries
+to load them one by one starting from the local ones until one snapshot is loaded correctly. If loading fails for all candidates,
+the node is started with empty database.
+
+To load a specific snapshot, `snapshots.snapshotsToLoad` parameter can be used. In that case the node searches for snapshots with
+block hash, provided in the parameter. Once again if loading of all such found files fails, the node is started with empty database.
+
+After a snapshot is loaded, earlier blocks (ones with smaller state index than snapshot) cannot be retrieved and committed to the DB
+(this is discussed in [Obtaining blocks section](#obtaining-blocks)). This constraint can cause problems (especially in case of reorg),
+if loaded snapshot is too recent. To avoid that, making snapshots is delayed by `snapshots.delay` states. E.g., if `snapshots.period`
+is `100` and `snapshots.delay` is `20`, then snapshot index `100` will be produced, when block index `120` is committed, snapshot index
+`200` will be produced, when snapshot index `220` is committed, etc... For the data to be available after this delay, `snapshot.delay`
+value must be considerably smaller than `stateManager.pruningMinStatesToKeep`.
```

Comment:
```suggestion
After a snapshot is loaded, earlier blocks (ones with a smaller state index than the snapshot) cannot be retrieved and committed to the DB
(this is discussed in [Obtaining blocks section](#obtaining-blocks)). This constraint can cause problems (especially in reorg)
if the loaded snapshot is too recent. To avoid that, making snapshots is delayed by `snapshots.delay` states. E.g., if `snapshots.period`
is `100` and `snapshots.delay` is `20`, then snapshot index `100` will be produced. When block index `120` is committed, snapshot index
`200` will be produced, when snapshot index `220` is committed, etc... For the data to be available after this delay, `snapshot.delay`
value must be considerably smaller than `stateManager.pruningMinStatesToKeep`.
```